### PR TITLE
Add Viewport Examples

### PR DESCRIPTION
**Pull request by macdrevx**
Ported from corresponding examples on Android ([ViewportShowcaseActivity.kt](https://github.com/mapbox/mapbox-maps-android/blob/main/app/src/main/java/com/mapbox/maps/testapp/examples/viewport/ViewportShowcaseActivity.kt), [AdvancedViewportGesturesExample.kt](https://github.com/mapbox/mapbox-maps-android/blob/main/app/src/main/java/com/mapbox/maps/testapp/examples/viewport/AdvancedViewportGesturesExample.kt)) with minor adjustments.

Also fixes an issue where the single tap and double tap to zoom in gestures could recognize simultaneously.

### CODE REVIEW

**Review by evil159 — APPROVED** [review]
Looking good! Maybe there can be some instructions on how to make these examples come to life. Not sure where to put them though. This could be beneficial especially for developers new to maps. I think having more detailed instructions here is more important than in other examples where the outcome is immediately visible. 

Things that can be pointed out:
- How to launch drive/run simulation in iOS simulator
- How to switch between viewport states in the advanced example

**Comment by macdrevx** [comment]
These are good suggestions. We do this sort of thing on docs.mapbox.com for certain examples.

For now, I'll add some comments.

File: Apps/Examples/Examples/All Examples/AdvancedViewportGesturesExample.swift [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        mapView.mapboxMap.setCamera(to: CameraOptions(center: cupertino, zoom: 14))
+
+        mapView.location.options.puckType = .puck2D(.makeDefault(showBearing: true))
+        mapView.location.options.puckBearingSource = .course
+
+        followPuckViewportState = mapView.viewport.makeFollowPuckViewportState(
+            options: FollowPuckViewportStateOptions(
+                bearing: .course))
+
+        overviewViewportState = mapView.viewport.makeOverviewViewportState(
```
Comment (pengdev) [inline Apps/Examples/Examples/All Examples/AdvancedViewportGesturesExample.swift:39]: seems you are using the real device's location for this example, while on Android, we have a custom LocationProvider implementation to emits fake location updates on a route. 
I think it is ok to be different on iOS, as the main showcase is how to customise gestures to work with the viewport, but would be good to consider adding the route simulator later on.

**Review by pengdev — APPROVED** [review]
LGTM with a comment.

File: Apps/Examples/Examples/All Examples/AdvancedViewportGesturesExample.swift (hunk 1 above)
Comment (macdrevx) [reply Apps/Examples/Examples/All Examples/AdvancedViewportGesturesExample.swift:39]: Added some comments to these examples explaining how to simulate location in the iOS simulator

**Pull request merged**